CMakeLists.txt

```diff
@@ -534,8 +534,14 @@ target_link_libraries(amazonCorrettoCryptoProvider ${PROBED_LINKED_FLAGS})
 # MacOS/Darwin because its gcc is actually an alias to clang, which does not
 # fully support them.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-   target_link_libraries(amazonCorrettoCryptoProvider -static-libstdc++)
-   target_link_libraries(amazonCorrettoCryptoProvider -static-libgcc)
+    message(STATUS "System name matches linux. Checking if -static-libstdc++ and -static-libgcc are supported by the compiler.")
+    CHECK_ENABLE_CXX_FLAG(CXX_SUPPORT_STATIC_LIBSTDCPP -static-libstdc++)
+    CHECK_ENABLE_CXX_FLAG(CXX_SUPPORT_STATIC_LIBGCC -static-libgcc)
+    if(CXX_SUPPORT_STATIC_LIBSTDCPP AND CXX_SUPPORT_STATIC_LIBGCC)
```

Comment:
This was a little strange. I witnessed three flavors of build issues: 1) sometime it would show as compiler issue with a message like "gcc internal", 2) CMake would configure aws-lc in its source directory rather than the build directory, 3) a runtime error like `undefined symbol: _GLOBAL__I__ZN108_GLOBAL__N__home_amirhosv_workplace_github_amazon_corretto_crypto_provider_csrc_loader.cpp_00000000_C4DE3DD010initializeEv`. When I searched, these symbols with absolute paths are for the sake of coverage calculation.
I don't think it's an LC issue: my hypothesis is that the issue is with our old CMake and GCC combination.